frontend/src/routes/candidate/(protected)/questions/[questionId]/+page.svelte

```diff
@@ -6,51 +6,157 @@
   import {HeadingGroup, PreHeading} from '$lib/components/headingGroup';
   import {LikertResponseButtons, QuestionActions, QuestionInfo} from '$lib/components/questions';
   import {BasicPage} from '$lib/templates/basicPage';
-  import {addAnswer, updateAnswer} from '$lib/api/candidate';
+  import {addAnswer, updateAnswer, deleteAnswer} from '$lib/api/candidate';
   import {get} from 'svelte/store';
-
-  const answers = get(answerContext.answers);
-  const answerStore = answerContext.answers;
+  import {onMount, onDestroy} from 'svelte';
+  import {candidateAppRoute} from '$lib/utils/routes';
 
   /**
    * A small delay before moving to the next question.
    * TODO: Make this a global variable used throughout the app.
    */
   const DELAY_M_MS = 350;
 
+  const SAVE_INTERVAL_MS = 1000;
+
+  let saveInterval: NodeJS.Timeout;
+  onMount(() => {
+    saveInterval = setInterval(() => {
+      saveOpenAnswerToLocal();
+    }, SAVE_INTERVAL_MS);
+  });
+
+  onDestroy(() => {
+    clearInterval(saveInterval);
+  });
+
+  const answers = get(answerContext.answers);
+  const answerStore = answerContext.answers;
+
+  $: questionId = $page.params.questionId;
+
+  // Local storage keys, depend on the question id
+  $: likertLocal = `candidate-app-question-${questionId}-likert`;
+  $: openAnswerLocal = `candidate-app-question-${questionId}-open`;
+
   let currentQuestion: QuestionProps | undefined;
   $: currentQuestion = $page.data.questions.find(
     (q) => q.id.toString() === $page.params.questionId.toString()
   );
 
-  $: answer = $answerStore[$page.params.questionId];
+  $: answer = $answerStore[$page.params.questionId]; // null if not answered
+
+  let selectedKey: AnswerOption['key'] | null = null;
+
+  // Local storage takes precedence over the answer store
+  $: {
+    const likertValue = localStorage.getItem(likertLocal);
+    selectedKey = likertValue ? parseInt(likertValue) : answer?.key ?? null;
+  }
+
+  let openAnswer = '';
+
+  // Load open answer from local storage and answer store if available, local storage takes precedence
+  $: localStorage.getItem(openAnswerLocal) && setOpenAnswer();
+  $: answer && setOpenAnswer();
+
+  function setOpenAnswer() {
+    if (answer && !localStorage.getItem(openAnswerLocal)) {
+      openAnswer = answer.openAnswer;
+      return;
+    }
+    openAnswer = localStorage.getItem(openAnswerLocal) ?? '';
+  }
+
+  function saveLikertToLocal({detail}: CustomEvent) {
+    selectedKey = detail.value;
+    localStorage.setItem(likertLocal, detail.value);
+  }
+
+  function saveOpenAnswerToLocal() {
+    if (openAnswer === '' || answer?.openAnswer === openAnswer) {
+      localStorage.removeItem(openAnswerLocal);
+      return;
+    }
+
+    localStorage.setItem(openAnswerLocal, openAnswer);
+  }
+
+  function removeLocalAnswers() {
```

Comment:
Matter of preference, but `removeLocalAnswers` makes me first assume that it would remove _all_ local answers saved (vs. just being limited to the current answer)